docs/zero_knowledge.md

```diff
@@ -1,136 +1,76 @@
+
+Now, using NIZKPs, instead of having to exchange multiple messages between $P$ and $V$, the former is able to prove knowledge of $w$ by sending just a single message. This is possible thanks to different approaches, like agreeing on a *Common Reference String* (CRS) [[2]](#2), or using the Fiat-Shamir heuristic [[3]](#3).
+
+We can see our NIZKP system as a blackbox with two sets of inputs, the statement $u$ being a set of public intputs, and the witness $w$ the set of private inputs. Such a blackbox generates a proof $\pi$ that $V$ will verify, as depicted here:
+<p align=center>
+<img src="images/zkp.png" width="50%"/>
+</p>
+
+## ZKP constructions
+
+Over time, different ZKP constructions have been designed. They have beautiful names like *STARK* (Scalable Transparent ARgument of Knowledge), *Bulletproof*, *SNARG* (Succinct Non-interactive ARGuments) and *SNARK* (Succinct Non-interactive ARgument of Knowledge). I will not explain the different protocols in detail here. Instead, I will just give a brief overview of what those fancy abbreviations actually mean so you have a starting point for the next internet research rabbit hole.
+
+- **Scalable:**  Both $P$ and $V$ running times are scalable.
+- **Transparent:** Doesn't require a setup-phase which uses a non-public random parameter like the shared reference string mentioned above.
+- **Argument:** Strictly mathematically speaking, many of the above protocols are no proofs but arguments. However in practice we can still think of them as a proofs.
+- **Knowledge:** The prover proofs that they themselves know the solution. In our above example that means that the prover doesn't prove that a solution exists but that they actually know the solution.
+- **Succinct:** The proofs are relatively small and easy to verify.
+- **Non-interactive:** No interaction between prover and verifier is needed.
 
 
 [graph coloring]: https://en.wikipedia.org/wiki/Graph_coloring#Vertex_coloring
-[graph img]: https://upload.wikimedia.org/wikipedia/commons/c/c2/Triangulation_3-coloring.svg
+
+## References
+<a id="1">[1]</a> 
+Goldwasser, S.; Micali, S.; Rackoff, C. "The knowledge complexity of interactive proof-systems". In Proceedings of the Seventeenth Annual ACM Symposium on Theory of Computing, Providence, RI, USA, 6–8 May 1985; ACM: New York, NY, USA, 1985; pp. 291–304.
+
+<a id="2">[2]</a> 
+Groth K.; Kohlweiss M.; Maller M.; Meiklejohn S.; Miers I. "Updatable and Universal Common Reference Strings with Applications to zk-SNARKs". Cryptology ePrint Archive, Report 2018/280.
+
+<a id="3">[3]</a> 
+Fiat, A.; Shamir, A. "How To Prove Yourself: Practical Solutions to Identification and Signature Problems", Advances in Cryptology --- CRYPTO' 86", 1987.
```

Comment:
I like how you included the references here! I tried before but couldn't figure out how to do it in a way that also renders on docs.rs. This solution works great though!